app/renderer/js/preload.js

```diff
@@ -0,0 +1,41 @@
+/*
+    This script will be loaded before other
+    scripts run in the webview. See the docs
+    for more info:
+
+    https://electron.atom.io/docs/api/webview-tag/#preload
+ */
+window.onload = () => {
+  const songs = (window.osts || []).concat(window.openings, window.endings);
```

Comment:
Unfortunately, we're pretty tightly coupled to the implementation in the web app here. Any changes to the underlying data structure will break this, but I was working with what I had and wanted to prevent making any changes to the web app just for this feature.

I think in the future you could consider emitting an event from the web app when a song is played to prevent having to care about implementation details here.

Reply:
What do you think will improve this?

Reply:
I think dispatching an event in your `play` function might do it. You could send the song metadata and listen for it here. Something worth investigating.

Reply:
Sounds good! Can you look into it later if you don't mind? 😄